README.md

```diff
@@ -317,6 +318,37 @@ The memory and the provided file store implementations both use a generic store
 When writing your own store implementation, you can do the same for APIs that don't need to do more than what the generic implementation provides.
 You can check [MemStore](https://github.com/nats-io/nats-streaming-server/blob/master/stores/memstore.go) and [FileStore](https://github.com/nats-io/nats-streaming-server/blob/master/stores/filestore.go) implementations for more details.
 
+## Store Encryption
+
+The server can be configured to encrypt messages payload when storing them, providing encryption at rest.
+This can be done from the command line or from the configuration file. Check `encrypt` and `encryption_key`
+in the [Configuring](#configuring) section.
+
+It is recommended to provide the encryption key through the environment variable `NATS_STREAMING_ENCRYPTION_KEY`
+instead of `encryption_key`. If encryption is enabled and `NATS_STREAMING_ENCRYPTION_KEY` is found, this
+will take precedence over `encryption_key` value.
+
+You can pass this from the command line this way:
+```
+$ env NATS_STREAMING_ENCRYPTION_KEY="mykey" nats-streaming-server -store file -dir datastore -encrypt
+```
+
+Note that only message payload is encrypted, all other data stored by NATS Streaming server is not.
+
+When running in clustering mode (see below), the server uses RAFT, which uses its own log files.
+Those will be encrypted too.
+
+Starting a server with `encrypt` against a datastore that was not encrypted will result in failures
+when it comes to decrypt a message, which may not happen right during the startup process. Instead,
+it will happen when attempting to deliver messages to consumers.
+
+The same behavior will occur if the encryption key is not the one that was used to encrypt the data.
```

Comment:
Should we have tooling to move a store from plaintext to encrypted?

Reply:
Going to be painful.. What I could do (and we should decide before accepting PR) is to had a few bytes at the beginning of an encrypted payload (before we copy the nonce) to indicate if the data is encrypted or not. So a store with encryption would look at the beginning of the data and if it does not find the marker return the payload as-is (assuming not encrypted). If the marker is present, decrypt. Obviously, if non-encrypted data starts with this marker, then this would result in decrypt failure.